app/drivers/adc.h

```diff
@@ -1,5 +1,9 @@
 #pragma once
 
 #include "stm32l4xx_hal.h"
+#include "errors.h"
 
-HAL_StatusTypeDef adc_init(void);
+#define ADC_INIT_FAIL	(ADC_BASE + 0)
+#define ADC_CONFIG_FAIL	(ADC_BASE + 1)
```

Comment:
Not sure about this FAIL suffix, maybe just do a E prefix like EADC_INIT

Reply:
Agreed. EADC_INIT etc is better.

Reply:
It's when HAL_ADC_Init fails, so can't really come up with a better name. EADC_HAL_INIT?